libevm/VM.cpp

```diff
@@ -153,14 +157,22 @@ void VM::checkRequirements(u256& io_gas, ExtVMFace& _ext, OnOpFunc const& _onOp,
 
 	case Instruction::CALL:
 	case Instruction::CALLCODE:
-		runGas = (bigint)m_schedule.callGas + m_stack[m_stack.size() - 1];
-		if (_inst != Instruction::CALLCODE && !_ext.exists(asAddress(m_stack[m_stack.size() - 2])))
+	case Instruction::DELEGATECALL:
+	{
+		runGas = (bigint)m_stack[m_stack.size() - 1];
+		if (_inst != Instruction::DELEGATECALL)
+			runGas += m_schedule.callGas;
+
+		if (_inst == Instruction::CALL && !_ext.exists(asAddress(m_stack[m_stack.size() - 2])))
 			runGas += m_schedule.callNewAccountGas;
-		if (m_stack[m_stack.size() - 3] > 0)
+
+		if (_inst != Instruction::DELEGATECALL && m_stack[m_stack.size() - 3] > 0)
 			runGas += m_schedule.callValueTransferGas;
-		newTempSize = std::max(memNeed(m_stack[m_stack.size() - 6], m_stack[m_stack.size() - 7]), memNeed(m_stack[m_stack.size() - 4], m_stack[m_stack.size() - 5]));
-		break;
 
+		unsigned sizesOffset = _inst == Instruction::DELEGATECALL ? 3 : 4;
+		newTempSize = std::max(memNeed(m_stack[m_stack.size() - sizesOffset - 2], m_stack[m_stack.size() - sizesOffset - 3]), memNeed(m_stack[m_stack.size() - sizesOffset], m_stack[m_stack.size() - sizesOffset - 1]));
```

Comment:
Suggestion: Break arguments to `std::max()` each one being in its own line. Also if we are using namespace std you can omit the namespace.